java/src/jmri/jmrit/operations/rollingstock/RollingStock.java

```diff
@@ -813,7 +815,7 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
                         if (e.getNewValue() != null) {
                             Location newLocation = locationManager.getLocationByReporter((jmri.Reporter) e.getNewValue());
                             if (newLocation != getLocation())
-                                setLocation(newLocation, null);
+                                setWhereLastSeen(newLocation);
```

Comment:
Hi Paul,

 

Sorry if I messed up your scripts.

 

The issue is that I have a user that wants to compare what the operations program believes the car is, versus what is actually provided by readers.  Previous to my latest change, when a car went through a reader, the code would update the car’s location, and null out the car’s track.  Then if the operations program detected train movement, would override the reader location, and update the car’s track.

 

The user asked that the operations car’s xml file provide him with both the location that operation believes the car is and what the reporters registered.  He wants to write his own program to find those discrepancies using the operations xml car file.

 

I was planning on adding more code that would highlight all cars in the table where a car’s operation location doesn’t match the tag’s last known location.  I’m also planning on logging all the “where last seen” location changes so a user can review how the car got to a particular location.

 

1) Why are you duplicating the When Last Seen and Where Last Seen properties of the ID Tag? Those should be persistent values associated with the ID tag, so we can look those up, and shouldn't need to be stored with the rolling stock.

My bad.  Didn’t realize that the tag was keeping those two things.  I see that now.

 

The user also wants to have a reader for each track in staging so he can have a wheel report by track.  So I’ll either need to save this info in operations, or add it to the ID tag.

 

2) The code that sets Where Last Seen and When Last Seen instead of the Location and the Date also need to store the Location and the Date. We may not have ID tag readers at ever location, but when a car passes a reader, that reader is associated with a location, and we know the car MUST be at that location.

I agree. But the two fields were being updated by two very different mechanisms.

The 2nd part of this is VERY important to me, because it changes the previous behaviour. This code change breaks scripts that I've got locally to print wheel reports.

Are you using operations and readers at the same time?  I don’t understand how this would work.

 

I would think if the ID tag has the car’s last known location and date, that your scripts could pull this from the car’s ID tag and not from the operation’s database.

 

Dan

 

From: Paul Bender [mailto:notifications@github.com] 
Sent: Wednesday, August 10, 2016 4:14 PM
To: JMRI/JMRI <JMRI@noreply.github.com>
Cc: Daniel Boudreau <daboudreau@hotmail.com>; State change <state_change@noreply.github.com>
Subject: Re: [JMRI/JMRI] Add to rolling stock RFID "where last seen" and "when last seen" (#1767)

 

In java/src/jmri/jmrit/operations/rollingstock/RollingStock.java <https://github.com/JMRI/JMRI/pull/1767#discussion_r74320994> :

> @@ -813,7 +815,7 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
>                          if (e.getNewValue() != null) {
>                              Location newLocation = locationManager.getLocationByReporter((jmri.Reporter) e.getNewValue());
>                              if (newLocation != getLocation())
> -                                setLocation(newLocation, null);
> +                                setWhereLastSeen(newLocation);

Dan, 

1) Why are you duplicating the When Last Seen and Where Last Seen properties of the ID Tag? Those should be persistent values associated with the ID tag, so we can look those up, and shouldn't need to be stored with the rolling stock.

2) The code that sets Where Last Seen and When Last Seen instead of the Location and the Date also need to store the Location and the Date. We may not have ID tag readers at ever location, but when a car passes a reader, that reader is associated with a location, and we know the car MUST be at that location.

The 2nd part of this is VERY important to me, because it changes the previous behaviour. This code change breaks scripts that I've got locally to print wheel reports.

Paul

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub <https://github.com/JMRI/JMRI/pull/1767/files/b8a4457a349fa99fa532337fd1e13bf1b2ad22c2#r74320994> , or mute the thread <https://github.com/notifications/unsubscribe-auth/ANw2p8mEpqzBl6QcjCNzosfdeDl1OjOzks5qejEKgaJpZM4JhdU1> .  <https://github.com/notifications/beacon/ANw2p51kOJNkhZZT8hQwBJxQRCndu3NNks5qejEKgaJpZM4JhdU1.gif>